### DISCUSSION TITLE
cross compilation requirements

### DISCUSSION THREAD

**Discussion by avin-hailo** [pybind/pybind11#5311] [Q&A]
The documentation is a bit lacking about this... What variables should be set during cross compilation?

I'm using a cmake toolchain file and I found that the result library is using the native processor name, even though my toolchain file defined the following:

```
set(CMAKE_SYSTEM_NAME Linux)
set(CMAKE_SYSTEM_PROCESSOR aarch64)

set(CMAKE_C_COMPILER aarch64-linux-gnu-gcc)
set(CMAKE_CXX_COMPILER aarch64-linux-gnu-g++)
set(CMAKE_STRIP aarch64-linux-gnu-strip CACHE FILEPATH "Strip")
```

To combat that, I defined `set(PYTHON_MODULE_EXTENSION ...)` using `CMAKE_SYSTEM_PROCESSOR`

Another issue is about python package. Is it enough to set `PYTHON_EXECUTABLE`? What should that value be when cross compling? I found that using the native (x64) python and cross compiling (aarch64) works - does that make sense?

So, to summarize - what cmake variables should I pass for a complete cross compilation
